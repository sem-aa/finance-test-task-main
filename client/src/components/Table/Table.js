import React from "react";
import id from "uniqid";
import {
  makeNameTickers,
  getDateFormat,
  getCurrentTime,
} from "../../helper/helper";
import { Table, Container } from "react-bootstrap";
import style from "./Table.module.css";
// import sprite from "../img/sprite.svg";

export default function TableTickers({ allTickers }) {
  const lastTickers = allTickers?.[allTickers.length - 1] || [];
  const prevTikers = allTickers?.[allTickers.length - 2] || [];

  const getPrevField = (tickerName, field) => {
    const exactTicker = prevTikers?.filter(
      (ticker) => ticker.ticker === tickerName
    );
    return exactTicker[0]?.[field];
  };

  const makeClass = (lastTicker, prevTicker) =>
    Number(lastTicker) > Number(prevTicker) ? style.green : style.red;

  return (
    <>
      <Container>
        <Table striped bordered hover>
          <thead>
            <tr>
              <th>Name</th>
              <th>Exchange</th>
              <th>Price</th>
              <th>Change</th>
              <th>Percent</th>
              <th>Dividend</th>
              <th>Yield</th>
              <th>Date</th>
              <th>Time</th>
            </tr>
          </thead>
          <tbody>
            {lastTickers?.map((ticker, index) => (
              <tr key={id()}>
                <>
                  <th>{makeNameTickers(ticker.ticker)} </th>
                  <th>{ticker.exchange} </th>
                  <th
                    className={makeClass(
                      ticker.price,
                      getPrevField(ticker.ticker, "price")
                    )}
                  >
                    {ticker.price} $
                  </th>
                  <th
                    className={makeClass(
                      ticker.change,
                      getPrevField(ticker.ticker, "change")
                    )}
                  >
                    {ticker.change}
                  </th>
                  <th
                    className={makeClass(
                      ticker.change_percent,
                      getPrevField(ticker.ticker, "change_percent")
                    )}
                  >
                    {ticker.change_percent}%{" "}
                  </th>
                  <th
                    className={makeClass(
                      ticker.dividend,
                      getPrevField(ticker.ticker, "dividend")
                    )}
                  >
                    {ticker.dividend}{" "}
                  </th>
                  <th
                    className={makeClass(
                      ticker.yield,
                      getPrevField(ticker.ticker, "yield")
                    )}
                  >
                    {ticker.yield}{" "}
                  </th>
                  <th>{getDateFormat(ticker.last_trade_time)}</th>
                  <th>{getCurrentTime(ticker.last_trade_time)}</th>
                </>
              </tr>
            ))}
          </tbody>
        </Table>
      </Container>
    </>
  );
}
