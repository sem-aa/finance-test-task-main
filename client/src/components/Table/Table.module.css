.green {
   

    color: green;
}

.red{
     color: red;
}

.down {
    height: 24px;
    width: 12px;
    fill: red;
}

.up {
    height: 24px;
    width: 12px;
    fill: green;
}