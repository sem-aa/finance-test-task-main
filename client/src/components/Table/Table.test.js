import React from "react";
import Enzyme, { shallow } from "enzyme";
import Table from "./Table";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

const mockTickers = [
  {
    ticker: "AAPL",
    exchange: "NASDAQ",
    price: "261.74",
    change: "166.56",
    change_percent: "0.06",
    dividend: "0.47",
    yield: "1.10",
    last_trade_time: "2021-09-15T07:23:06.000Z",
  },
  {
    ticker: "GOOGL",
    exchange: "NASDAQ",
    price: "185.40",
    change: "170.00",
    change_percent: "0.19",
    dividend: "0.41",
    yield: "1.83",
    last_trade_time: "2021-09-15T07:23:41.000Z",
  },
];

describe("Table component", () => {
  test("render", () => {
    const wrapper = shallow(<Table tickers={mockTickers} />);

    expect(wrapper).toMatchSnapshot();
  });
});

test("returns the default emty array when there is no tickers to map through", () => {
  const wrapper = shallow(<Table />);

  expect(wrapper).toMatchSnapshot();
});




