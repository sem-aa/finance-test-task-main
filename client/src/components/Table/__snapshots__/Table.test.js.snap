// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component render 1`] = `ShallowWrapper {}`;

exports[`returns the default emty array when there is no tickers to map through 1`] = `ShallowWrapper {}`;
